### ISSUE TITLE
Zero Gradient from Feature Loss

### ISSUE THREAD

**Issue by arishov1** [yuval-reshef/StreamVC#3]
Hi,

I think we should perform detach only on `real[i][j]`, not on the difference between real and fake (`torch.abs(fake[i][j] - real[i][j])`).

Before:
```

class FeatureLoss(nn.Module):
    def __init__(self, n_blocks=3, n_features=16, n_layers=4):
        super().__init__()
        self.n_blocks = n_blocks
        self.n_features = n_features
        self.n_layers = n_layers

    def forward(self, real, fake, mask_ratio):
        loss = torch.tensor(
            0., device=real[0][0].device, dtype=real[0][0].dtype)
        feature_weights = 4.0 / (self.n_layers + 1)
        discriminator_weights = 1.0 / self.n_blocks
        wt = discriminator_weights * feature_weights
        for i in range(self.n_blocks):
            for j in range(len(fake[i]) - 1):
                loss += wt * \
                    masked_mean_from_ratios(
                        torch.abs(fake[i][j]-real[i][j]).detach()), mask_ratio)
        return loss
```
After:
```
class FeatureLoss(nn.Module):
    def __init__(self, n_blocks=3, n_features=16, n_layers=4):
        super().__init__()
        self.n_blocks = n_blocks
        self.n_features = n_features
        self.n_layers = n_layers

    def forward(self, real, fake, mask_ratio):
        loss = torch.tensor(
            0., device=real[0][0].device, dtype=real[0][0].dtype)
        feature_weights = 4.0 / (self.n_layers + 1)
        discriminator_weights = 1.0 / self.n_blocks
        wt = discriminator_weights * feature_weights
        for i in range(self.n_blocks):
            for j in range(len(fake[i]) - 1):
                loss += wt * \
                    masked_mean_from_ratios(
                        torch.abs(fake[i][j]-real[i][j].detach()), mask_ratio) # doing detach only on real part
        return loss
```

**Comment by yuval-reshef** [comment]
looks like you are correct. I pushed this change to main
